src/main/java/decoder/Decoder.java

```diff
@@ -14,16 +14,19 @@ public static void decode(InputStream dataStream, OutputStream outputStream) {
     // data file and include headers in the proto definitions so we could
```

Comment:
would the parameters change from taking a input and output stream to taking a reference for the file wanted from storage?